srml/support/src/storage/mod.rs

```diff
@@ -19,97 +19,21 @@
 use crate::rstd::prelude::*;
 use crate::rstd::borrow::Borrow;
 use codec::{Codec, Encode, Decode, KeyedVec, EncodeAppend};
-use hashed::generator::{HashedStorage, StorageHasher};
-use unhashed::generator::UnhashedStorage;
 
 #[macro_use]
 pub mod storage_items;
 pub mod unhashed;
 pub mod hashed;
-
-/// The underlying runtime storage.
-pub struct RuntimeStorage;
-
-impl<H: StorageHasher> HashedStorage<H> for RuntimeStorage {
-	fn exists(&self, key: &[u8]) -> bool {
-		hashed::exists(&H::hash, key)
-	}
-
-	/// Load the bytes of a key from storage. Can panic if the type is incorrect.
-	fn get<T: Decode>(&self, key: &[u8]) -> Option<T> {
-		hashed::get(&H::hash, key)
-	}
-
-	/// Put a value in under a key.
-	fn put<T: Encode>(&mut self, key: &[u8], val: &T) {
-		hashed::put(&H::hash, key, val)
-	}
-
-	/// Remove the bytes of a key from storage.
-	fn kill(&mut self, key: &[u8]) {
-		hashed::kill(&H::hash, key)
-	}
-
-	/// Take a value from storage, deleting it after reading.
-	fn take<T: Decode>(&mut self, key: &[u8]) -> Option<T> {
-		hashed::take(&H::hash, key)
-	}
-
-	fn get_raw(&self, key: &[u8]) -> Option<Vec<u8>> {
-		hashed::get_raw(&H::hash, key)
-	}
-
-	fn put_raw(&mut self, key: &[u8], value: &[u8]) {
-		hashed::put_raw(&H::hash, key, value)
-	}
-}
-
-impl UnhashedStorage for RuntimeStorage {
-	fn exists(&self, key: &[u8]) -> bool {
-		unhashed::exists(key)
-	}
-
-	/// Load the bytes of a key from storage. Can panic if the type is incorrect.
-	fn get<T: Decode>(&self, key: &[u8]) -> Option<T> {
-		unhashed::get(key)
-	}
-
-	/// Put a value in under a key.
-	fn put<T: Encode + ?Sized>(&mut self, key: &[u8], val: &T) {
-		unhashed::put(key, val)
-	}
-
-	/// Remove the bytes of a key from storage.
-	fn kill(&mut self, key: &[u8]) {
-		unhashed::kill(key)
-	}
-
-	/// Remove the bytes of a key from storage.
-	fn kill_prefix(&mut self, prefix: &[u8]) {
-		unhashed::kill_prefix(prefix)
-	}
-
-	/// Take a value from storage, deleting it after reading.
-	fn take<T: Decode>(&mut self, key: &[u8]) -> Option<T> {
-		unhashed::take(key)
-	}
-
-	fn get_raw(&self, key: &[u8]) -> Option<Vec<u8>> {
-		unhashed::get_raw(key)
-	}
-
-	fn put_raw(&mut self, key: &[u8], value: &[u8]) {
-		unhashed::put_raw(key, value)
-	}
-}
+pub mod child;
+pub mod generator;
 
 /// A trait for working with macro-generated storage values under the substrate storage API.
 pub trait StorageValue<T: Codec> {
 	/// The type that get/take return.
 	type Query;
 
 	/// Get the storage key.
-	fn key() -> &'static [u8];
+	fn key() -> Vec<u8>;
```

Comment:
Why does this return a `Vec<u8>`?

Reply:
the PR changes the key function to returned hashed key instead of raw keys. thus this is not static anymore.

Instead of Vec we could return `[u8; 32]` as we always hash StorageValue with twox_128.

Reply:
Sounds like a good idea.

Reply:
done